arkouda/categorical.py

```diff
@@ -529,3 +534,197 @@ def concatenate(self, others : List[Categorical], ordered : bool=True) -> Catego
                                   ordered=ordered)
             newvals = wherediditgo[oldvals]
             return Categorical.from_codes(newvals, newidx)
+
+    def _get_vars_with_register(self) -> List[Tuple[str, Union[pdarray, Strings]]]:
+        """
+        Internal method to retrieve the components of this object which are able to be registered
+
+        Returns
+        -------
+        List[Tuple(str, Union[pdarray, Strings])]
+            List of tuples
+        """
+        return [(k, v) for k, v in vars(self).items()
+                if k in self.RegisterableComponents and hasattr(v, "register")]
+
+    @typechecked()
+    def register(self, user_defined_name:str) -> Categorical:
+        """
+        Register this Categorical object and underlying components with the Arkouda server
+
+        Parameters
+        ----------
+        user_defined_name : str
+            user defined name the Categorical is to be registered under
+            this will be the root name for underlying components
+
+        Returns
+        -------
+        Categorical
+            The same Categorical which is now registered with the arkouda server and has an updated name.
+            This is an in-place modification, the original is returned to support a fluid programming style.
+            Please note you cannot register two different Categoricals with the same name.
+
+        Raises
+        ------
+        TypeError
+            Raised if user_defined_name is not a str
+        RegistrationError
+            If the server was unable to register the Categorical with the user_defined_name
+            If the Categorical object has no underlying register-able components.
+
+        See also
+        --------
+        unregister, attach, unregister_categorical_by_name, is_registered
+
+        Notes
+        -----
+        Objects registered with the server are immune to deletion until
+        they are unregistered.
+        """
+        components = self._get_vars_with_register()
+        if components:
+            for k, v in components:
+                self.logger.debug(f"Registering attribute {k} with name {user_defined_name}_{k}")
```

Comment:
i noticed you use a `.` on the next line vs a `_` to separate the field name, we should be consistent. Strings uses a `_` I think

Reply:
We chatted and decided to use the dot character for all sub-components (so changes to Categorical & Strings).

Note: I updated the #761 ticket with the items we discussed.